lib/assert.js

```diff
@@ -518,13 +530,11 @@ function partiallyCompareArrays(actual, expected, comparedObjects) {
 
   // Create a map to count occurrences of each element in the expected array
   const expectedCounts = new SafeMap();
-  const safeExpected = new SafeArrayIterator(expected);
 
-  for (const expectedItem of safeExpected) {
-    // Check if the item is a zero or a -0, as these need to be handled separately
+  for (const expectedItem of new SafeArrayIterator(expected)) {
```

Comment:
If we're looking for performance, we should probably be using a classic `for(;;)` loop. I'd recommend rolling back this change (i.e. put back the `safeExpected` assignment, because it has nothing to do with performance) for this PR, and make a new one switching to a `for(;;)` loop on which we can run benchmarks

Reply:
if you do this tho you would have to call the `ArrayFrom` primordial. The performance enhancement was to adjust the "low hanging fruit" stuff I could get, "structure" wise, without influencing readability.  IMHO the current implementation is better, but happy to change if you don't agree!

reverted the assignment tho